### PR TITLE
feat: oauth login

### PR DESCRIPTION
**Pull request by eldu**
- Adds Oauth, restrict to brown.edu
- Makes add pub only available after logging in
- refactors nav bar

After this PR goes through need to go back and edit the firebase rules to the following to ensure only logged in users can write to the firebase
```
    match /publications/{document=**} {
      allow read: if true;
      allow write: request.auth != null;
    }
```

⚠️ This PR doesn't restrict to a CCV group. Only to brown.edu

The following code would have let us see a user's groups. However, I hit a 403. **It's possible that the Google Workspace administrator has not granted the necessary permissions for your application to access the groups.**

```js
// provider.addScope(
//   "https://www.googleapis.com/auth/admin.directory.group.readonly"
// );
const collectionName = 'publications';

export const handleLogin = async () => {
  try {
    await signInWithPopup(auth, provider);
    // const credential = GoogleAuthProvider.credentialFromResult(result)
    // const token = credential.accessToken
    // const user = result.user

    // const response = await fetch(`http://admin.googleapis.com/admin/directory/v1/groups?userKey=${encodeURIComponent(user.email)}`, {
    //     headers: {
    //       'Authorization': `Bearer ${token}`
    //     }
    // })
```

### CODE REVIEW

**Comment by eldu** [comment]
Opened an issue to investigate scope
https://github.com/brown-ccv/ccv-pubs-frontend/issues/67

File: src/utils/firebase.ts [hunk 1]
```diff
@@ -26,9 +33,61 @@ const firebaseConfig = {
...
+ */
+export const useAuthStateChanged = () => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth, (user) => {
+      if (user) {
+        const { displayName, email, uid } = user;
+
+        console.log(user);
```
Comment (hetd54) [inline src/utils/firebase.ts:71]: Do we want to keep this log?

File: src/styles/custom.scss [hunk 2]
```diff
@@ -1,12 +1,11 @@
 // Import Bootstrap
 @import '~bootstrap/scss/bootstrap.scss';
-@import '~brown-university-styles/dist/styles.css';
+//@import '~brown-university-styles/dist/styles.css';
```
Comment (hetd54) [inline src/styles/custom.scss:3]: Do you want to remove?

**Review by hetd54 — APPROVED** [review]
Gorgeous! Tested with brown email, and confirmed block with personal account.

**Pull request merged**